fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.tsx

```diff
@@ -473,6 +473,129 @@ export class FmcAircraftInterface {
     }, 200);
   }
 
+  /*
+        When the aircraft is in the holding, predictions assume that the leg is flown at holding speed
+        with a vertical speed equal to - 1000 ft/mn until reaching a restrictive altitude constraint, the
+        FCU altitude or the exit fix. If FCU or constraint altitude is reached first, the rest of the
+        pattern is assumed to be flown level at that altitude
+        */
+  getHoldingSpeed(speedConstraint = null, altitude = null) {
+    const fcuAltitude = SimVar.GetSimVarValue('AUTOPILOT ALTITUDE LOCK VAR:3', 'feet');
+    const alt = Math.max(fcuAltitude, altitude ? altitude : 0);
+
+    let kcas = SimVar.GetSimVarValue('L:A32NX_SPEEDS_GD', 'number');
+    if (this.fmgc.getFlightPhase() === FmgcFlightPhase.Approach) {
+      kcas = this.getAppManagedSpeed();
+    }
+
+    if (speedConstraint != null && speedConstraint > 100) {
+      kcas = Math.min(kcas, speedConstraint);
+    }
+
+    // apply icao limits
+    if (alt < 14000) {
+      kcas = Math.min(230, kcas);
+    } else if (alt < 20000) {
+      kcas = Math.min(240, kcas);
+    } else if (alt < 34000) {
+      kcas = Math.min(265, kcas);
+    } else {
+      kcas = this.fmgc.guidanceController?.atmosphericConditions.computeCasFromMach(alt, 0.83);
```

Comment:
This is getting well beyond the Law of Demeter 😬 .

Reply:
Indeed, but I didn't find a good alternative. Duplicating the code like in the original code is error prone, and creating an interface at the top of the chain just clutters the classes. Any idea?